src/bundle/Resources/public/js/scripts/admin.anchor.navigation.js

```diff
@@ -0,0 +1,63 @@
+(function(global, doc, eZ) {
+    const sectionsPosition = {};
+    const scrollOffset = 400;
+    const formContainerNode = doc.querySelector('.ibexa-content-edit-content');
+    const isVerticalScrollVisible = () => {
+        const { scrollHeight, offsetHeight } = formContainerNode;
+
+        return scrollHeight > offsetHeight;
+    };
+    const showSection = (sectionId) => {
+        doc.querySelectorAll('.ibexa-anchor-navigation-menu__btn').forEach((btn) => {
+            const { anchorTargetSectionId } = btn.dataset;
+
+            btn.classList.toggle('ibexa-anchor-navigation-menu__btn--active', anchorTargetSectionId === sectionId);
+        });
+
+        doc.querySelectorAll('.ibexa-anchor-navigation-sections__section').forEach((section) => {
+            const { anchorSectionId } = section.dataset;
+
+            section.classList.toggle('ibexa-anchor-navigation-sections__section--active', anchorSectionId === sectionId);
+        });
+    };
+    const navigateTo = (event) => {
+        let startSection = 0;
+        const { anchorTargetSectionId } = event.currentTarget.dataset;
+
+        doc.querySelectorAll('.ibexa-anchor-navigation-sections__section').forEach((section) => {
+            const { anchorSectionId } = section.dataset;
+
+            sectionsPosition[anchorSectionId] = startSection;
+            startSection = section.offsetHeight;
+        });
+
+        if (isVerticalScrollVisible()) {
+            formContainerNode.scrollTo({
+                top: sectionsPosition[anchorTargetSectionId],
+                behavior: 'smooth',
+            });
+        } else {
+            showSection(anchorTargetSectionId);
+        }
+    };
+
+    doc.querySelectorAll('.ibexa-anchor-navigation-menu__btn').forEach((btn) => {
+        btn.addEventListener('click', navigateTo, false);
+    });
+
+    if (formContainerNode) {
+        formContainerNode.addEventListener('scroll', () => {
+            const position = formContainerNode.scrollTop + scrollOffset;
+
+            doc.querySelectorAll('.ibexa-anchor-navigation-sections__section').forEach((section) => {
+                const { anchorSectionId } = section.dataset;
+                const start = section.offsetTop;
+                const end = section.offsetHeight + section.offsetTop;
+
+                if (position > start && position < end) {
+                    showSection(anchorSectionId);
+                }
+            });
```

Comment:
From my perspective, you basically try to find one and only one section and then call `showSection` on it, so in my opinion it would be more expressive to do it this way:
```suggestion
            const allSections = doc.querySelectorAll('.ibexa-anchor-navigation-sections__section');
            const activeSection = allSelections.find((section) => {
                const { anchorSectionId } = section.dataset;
                const start = section.offsetTop;
                const end = section.offsetHeight + section.offsetTop;

                return position > start && position < end;
            });
            const activeSectionId = activeSection.dataset.anchorSectionId
            
            showSection(activeSectionId);
```